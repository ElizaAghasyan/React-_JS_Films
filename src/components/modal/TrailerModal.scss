@import 'src/scss/mixin';
@import 'src/scss/variables';

.modal {
  display: flex;
  justify-content: center;
  position: fixed;
  z-index: 99;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  overflow: auto;
  opacity: 0;
  visibility: hidden;

  &_active {
    opacity: 1;
    visibility: visible;
  }

  &__content {
    border-radius: 2em;
    position: absolute;
    top: 1rem;
    width: 98%;
    background: $black;
    padding: 2rem;
  }
}
