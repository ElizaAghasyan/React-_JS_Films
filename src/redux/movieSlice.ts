import {createAsyncThunk, createSlice} from "@reduxjs/toolkit";
import serverApi, {movieType} from "../config/movieApi";

const initialState = {
    movies: {},
}

interface MovieType  {
    id: number;
    title: string;
    overview: string;
    poster_path: string;
    vote_average: number;
    backdrop_path: string;
    vote_count: number;
    name: string;
}


export const fetchAsyncMovies = createAsyncThunk('movies/fetchAsyncMovies', async () => {
    const params = {page: 1}
    let response = await serverApi.getMovieList(movieType.popular, {params})
        .catch(err => {
            console.log(err)
        })
    let result = await response?.data.results[Math.floor(Math.random() * response.data.results.length - 1)]
    return result;
});


export const movieSlice = createSlice({
    name: 'movies',
    initialState,
    reducers: {
        addMovies: (state, {payload}) => {
            state.movies = payload
        }
    },
    extraReducers: (builder) => {
        builder.addCase(fetchAsyncMovies.pending, (state, action) => {
            console.log("Pending...")
        })
        builder.addCase(fetchAsyncMovies.fulfilled, (state: {}, action) => {
            return {
                ...state, movies: action.payload
            }
        })
        builder.addCase(fetchAsyncMovies.rejected, (state, action) => {
            console.log("Pending...")
        })
    }
});

export const {addMovies} = movieSlice.actions;

export const getBannerMovie = (state: { movies: { movies: MovieType }}) => state.movies.movies;


export default movieSlice.reducer;
