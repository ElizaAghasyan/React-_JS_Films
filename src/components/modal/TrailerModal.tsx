import React, { useEffect, useRef } from "react";
import movieApi, { category } from "../../config/movieApi";

import './TrailerModal.scss';
import CloseIcon from "@mui/icons-material/Close";
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles({
    close: {
        position: 'absolute',
        fontSize: '2.5rem',
        top: '1.1rem',
        right: '1rem',
        color: '#fefefe',
        cursor: 'pointer'
    }
});

type TrailerModalType = {
    active: boolean;
    id: number
};

const TrailerModal = ({ active, id }: TrailerModalType ) => {
    const iframeRef = useRef<HTMLIFrameElement>(null);
    const modalRef = useRef<HTMLIFrameElement>(null);
    const classes = useStyles();

    useEffect(() => {
        const getVideos = async () => {
            let  videos;
            if(id) {
                videos = await movieApi.getVideos(category.movie, id);
            }

            if (videos?.data.results && videos.data.results.length > 0) {
                const videSrc = 'https://www.youtube.com/embed/' + videos.data.results[0].key;
                iframeRef.current && iframeRef.current.setAttribute('src', videSrc);
            }
        }
        getVideos()
    }, )

    const onClose = () => {
        iframeRef.current && iframeRef.current.setAttribute('src', '');
        modalRef.current && modalRef.current.classList.remove('modal_active');
    }

    return (
        <div ref={modalRef} className={active ? `modal modal_active` : `modal`}>
            <div className="modal__content">
                <iframe ref={iframeRef} width="100%" height="450px" title="trailer" frameBorder="0" >
                </iframe>
            </div>
            <CloseIcon onClick={onClose} className={classes.close} />
        </div>
    )
}

export default TrailerModal;
