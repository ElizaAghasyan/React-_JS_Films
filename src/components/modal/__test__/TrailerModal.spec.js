import TrailerModal from "../TrailerModal";
import renderer from "react-test-renderer";
import {store} from "../../../redux/store";
import {Provider} from "react-redux";

describe('TrailerModal Component', () => {

    it('TrailerModal snapshot test', () => {
        const component = renderer.create(<Provider store={store}><TrailerModal/></Provider>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
