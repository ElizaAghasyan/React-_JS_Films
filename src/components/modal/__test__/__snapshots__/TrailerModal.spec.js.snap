// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailerModal Component TrailerModal snapshot test 1`] = `
<div
  className="modal"
>
  <div
    className="modal__content"
  >
    <iframe
      frameBorder="0"
      height="450px"
      title="trailer"
      width="100%"
    />
  </div>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-close-1 css-i4bv87-MuiSvgIcon-root"
    data-testid="CloseIcon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</div>
`;
